docs/pages/guides/schema-validation.mdx

```diff
@@ -0,0 +1,120 @@
+
+## Create a schema
+To create a schema, you will need to POST to the schemas endpoint:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+{
+  "id": "some-id",
+  "name": "Your schema name",
+  "data": `{
+    type Storage {
+      // your data schema will go here
+    }
+  }`,
+}
+
+```
+Note that the data type will be of type string because the schema will be parsed. For more information, see our [Schema Language documentation](xxx).
+The operation above creates the schema, but no schema is attached to a room. To attach this schema to a room, you will POST to the rooms endpoint:
+```ts
+POST https://api.liveblocks.io/v2/rooms/{id}/schema
+{
+  "versionId": "some-id"
+}
+```
+## The validation process in practice
+If you would like to work through an example, you can follow along by using the Multiplayer Form example.
+
+The schema we will define for the example is the following:
+```ts
+POST https://api.liveblocks.io/v2/schemas
+
+{
+  "id": "collaborative-form",
+  "name": "Collaborative Form",
+  "data": `
+    type Logo {
+      name: String
+      theme: String
+    }
+    type storage {
+      logo: LiveObject<Logo>
+    }
+  `
+}
+// 200 Response
+{
+   "versionId": "collaborative-form@1",
+   "schemaId": "collaborative-form",
+   "name": "Collaborative Form",
+   "createdAt": "2023-01-22T15:10:25.225Z",
+   "updatedAt": "2023-01-22T15:10:25.225Z"
+}
+```
+
+To attach this schema to a room, you will POST to the rooms endpoint:
+
+```ts
+POST https://api.liveblocks.io/v2/rooms/{id}/schema
+{
+  "schema": "collaborative-form@1"
+}
+```
+
+To simulate the importance of schema validation and also demonstrate how operations can fail, we will change the nested value of `name` to `organization`:
+
+```tsx file="_app.tsx"
+    <RoomProvider
+      id={roomId}
+      initialPresence={{ focusedId: null }}
+      initialStorage={{
+        logo: new LiveObject({
+          organization: "Acme Inc.",
+          theme: "light",
+        }),
+      }}
+    >
+```
+In `index.tsx`, we will update all references from `name` to `organization`: this is a simple but common instance of updating variable names, which could easily break our integration. 
+After updating and running locally, we should now see an error because our deployed schema and client code are out of sync.
+
+<!----Insert sample failure message here, once this is released. You also mentioned demonstrating a refresh, but I have no way to record this currently !--->
```

Comment:
@nvie @BitPhinix: I didn't have time to review the latest PR about rejections, but would it be possible to provide a screenshot or the error message corresponding to this issue? 🙏

Reply:
This would be great 👍

Reply:
Here is a screenshot I faked locally. I've made a video and a static screen shot. Use whichever one you like best. I think the animated version can help to see the interaction in an app.

The error message itself is completely faked and will likely change a lot in the next couple of weeks.

**Video:**
https://user-images.githubusercontent.com/83844/216575242-18933270-8ae7-4ac1-9a36-c2a78b4717c9.mp4

**Static screenshot:**
<img width="1276" alt="Screen Shot 2023-02-03 at 11 22 46@2x" src="https://user-images.githubusercontent.com/83844/216576155-13a85093-0ce2-400d-bf22-1dd1852f559b.png">

Reply:
Ugh. I copy/pasted the incorrect fake error message that doesn't make sense in the spreadsheet app. I'll redo the shots.

Reply:
Here is a screenshot I faked locally. I've made a video and a static screen shot. Use whichever one you like best. I think the animated version can help to see the interaction in an app.

The error message itself is completely faked and will likely change a lot in the next couple of weeks.

**Video:**

https://user-images.githubusercontent.com/83844/216578151-6ae25bdc-9c17-4d2d-80e1-ae3e212a8440.mp4

**Static screenshot:**

<img width="1320" alt="Screen Shot 2023-02-03 at 11 29 46@2x" src="https://user-images.githubusercontent.com/83844/216577786-7ee54263-2355-491b-86f6-6c4df107974a.png">